functorch/_src/aot_autograd.py

```diff
@@ -567,7 +1105,106 @@ def backward(ctx, *flat_args):
 
     @wraps(CompiledFunction.apply)
     def compiled_function(*args):
-        return CompiledFunction.apply(*remove_dupe_args(args))
+        # Step 1: remove dupe args
+        no_dupe_args = remove_dupe_args(args)
+
+        # Step 2: remove aliased inputs that are mutated, replace with synthetic bases
+        # Only happens if our graph mutates an input that aliases another input.
+        if CompiledFunction.synthetic_base_info is not None:
+            # Given: the original args, including at least one pair of inputs that are aliased
+            # and get subsequently mutated.
+            # Generate: the updated args, including (potentially multiple) synthetic bases
+            # that replace the views. The input views are regenerated manually in the compiled function.
+            # TODO: think harder about what happens if (a view of) one of these mutated input views is ALSO returned
+            new_inputs, metadata = merge_view_inputs(no_dupe_args, CompiledFunction.fw_metadata.mutated_input_info)
```

Comment:
So `merge_view_inputs` is hotpath. We may need to more carefully audit it for speed

Reply:
Agreed - I didn't try to carefully optimize it for now just because I think the "mutate an input that aliases another input" is pretty uncommon, but this is probably worth coming back to.